packages/jellyfish-api-core/__tests__/category/masternode/updateMasternode.test.ts

```diff
@@ -0,0 +1,838 @@
+  const testing = Testing.create(container)
+
+  beforeAll(async () => {
+    await testing.container.start()
+    await testing.container.waitForWalletCoinbaseMaturity()
+
+    await testing.generate(9) // Generate 9 blocks to move to block 110
+
+    const blockCount = await testing.rpc.blockchain.getBlockCount()
+    expect(blockCount).toStrictEqual(110) // At greatworldheight
+  })
+
+  afterAll(async () => {
+    await testing.container.stop()
+  })
+
+  async function waitUntilMasternodePreEnabled (masternodeId: string, timeout = 100000): Promise<void> {
+    return await waitForCondition(async () => {
+      const data: MasternodeResult<MasternodeInfo> = await testing.container.call('getmasternode', [masternodeId])
+
+      if (Object.values(data)[0].state !== MasternodeState.PRE_ENABLED) {
+        await testing.generate(1)
+        return false
+      }
+      return true
+    }, timeout, 100, 'waitUntilMasternodePreEnabled')
+  }
+
+  async function waitUntilMasternodeEnabled (masternodeId: string, timeout = 100000): Promise<void> {
+    return await waitForCondition(async () => {
+      const data: MasternodeResult<MasternodeInfo> = await testing.container.call('getmasternode', [masternodeId])
+
+      if (Object.values(data)[0].state !== MasternodeState.ENABLED) {
+        await testing.generate(1)
+        return false
+      }
+      return true
+    }, timeout, 100, 'waitUntilMasternodeEnabled')
+  }
+
+  it('should updateMasternode', async () => {
+    // Several updateMasternode calls within different blocks
+    {
+      const masternodeId1 = await testing.container.call('createmasternode', [await testing.generateAddress()])
+      await testing.generate(1)
+
+      const masternodeId2 = await testing.container.call('createmasternode', [await testing.generateAddress()])
+      await testing.generate(1)
+
+      const masternodeId3 = await testing.container.call('createmasternode', [await testing.generateAddress()])
+      await testing.generate(1)
+
+      const masternodeId4 = await testing.container.call('createmasternode', [await testing.generateAddress()])
+      await testing.generate(1)
+
+      const masternodeId5 = await testing.container.call('createmasternode', [await testing.generateAddress()])
+      await testing.generate(1)
+
+      const masternodeId6 = await testing.container.call('createmasternode', [await testing.generateAddress()])
+      await testing.generate(1)
+
+      const masternodeId7 = await testing.container.call('createmasternode', [await testing.generateAddress()])
+      await testing.generate(1)
+
+      await waitUntilMasternodeEnabled(masternodeId1)
+      await waitUntilMasternodeEnabled(masternodeId2)
+      await waitUntilMasternodeEnabled(masternodeId3)
+      await waitUntilMasternodeEnabled(masternodeId4)
+      await waitUntilMasternodeEnabled(masternodeId5)
+      await waitUntilMasternodeEnabled(masternodeId6)
+      await waitUntilMasternodeEnabled(masternodeId7)
```

Comment:
I feel like this one line =>  `await waitUntilMasternodeEnabled(masternodeId7)` is enough, but I still add waiting for all 7 masternodes code to increase readibility